pkg/jsonmessage/jsonmessage.go

```diff
@@ -189,13 +189,9 @@ func DisplayJSONMessagesStream(in io.Reader, out io.Writer, terminalFd uintptr,
 				if isTerminal {
 					fmt.Fprintf(out, "\n")
 				}
-			} else {
-				diff = len(ids) - line
 			}
-			if isTerminal {
-				// NOTE: this appears to be necessary even if
-				// diff == 0.
-				// <ESC>[{diff}A = move cursor up diff rows
+			diff = len(ids) - line
```

Comment:
In fact we explicitly initialise `diff := 0` above (way outside the context and scope shown here) but the line assigning to `diff` here is now outside the `if !ok` altogether, including being outside the `else` case, nothing inside any part of the `if` touches `diff` in any way any more.

I don't think the fact that `diff` is `0` up until the point that we assign another value is relevant enough to comment upon, the previous value may as well be arbitrary as far as this block is concerned.